remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/RemoteMessagingConfigDownloadScheduler.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.remote.messaging.impl
+
+import android.content.Context
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.work.*
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesMultibinding
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class RemoteMessagingConfigDownloadWorker(context: Context, workerParameters: WorkerParameters) : CoroutineWorker(context, workerParameters) {
+    lateinit var downloader: RemoteMessagingConfigDownloader
+    lateinit var dispatcherProvider: DispatcherProvider
+
+    override suspend fun doWork(): Result {
+        return withContext(dispatcherProvider.io()) {
+            val result = downloader.download()
+            return@withContext if (result) {
+                Result.success()
+            } else {
+                Result.retry()
+            }
+        }
+    }
+}
+
+@ContributesMultibinding(AppScope::class)
+class RemoteMessagingConfigDownloadScheduler @Inject constructor(
+    private val workManager: WorkManager
+) : LifecycleObserver {
+
+    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
```

Comment:
Not my PR to review but I saw this the other day 😅 . `OnLifecycleEvent` is not deprecated and we should be using `LifecycleEventObserver` instead (we already are in some places).